### ISSUE TITLE
Add dedicated Deepseek model provider

### ISSUE THREAD

**Issue by GabrielBianconi** [tensorzero/tensorzero#483]
> [!TIP]
>
> You can already use TensorZero + Deepseek using the [OpenAI-compatible model provider](https://www.tensorzero.com/docs/gateway/guides/providers/openai-compatible).

---

Let's add a dedicated inference provider for Deepseek (`provider_type = "deepseek"`).

* Create a new provider in `gateway/src/inference/providers`
* Add the necessary entries to `gateway/src/model.rs` and `gateway/src/error.rs`
* Add your provider to the E2E test suite: `gateway/tests/e2e/providers`
* Add a minimal example to `examples/guides/providers`
* As the codebase evolves, providers are references elsewhere as well. Search for `mistral` and see wherever it is mentioned. Chances are there should be a similar mention of `deepseek`.
 
See #529 and #547 for recent provider integration PRs.

**Comment by ankit-varma10** [comment]
@GabrielBianconi This is a great project to understand the inner working of tensorzero. Could you please assign me this issue? Thanks!

**Comment by GabrielBianconi** [comment]
@ankit-varma10 Sounds good, thanks!

**Comment by GabrielBianconi** [comment]
Hi @ankit-varma10, are you still planning to submit a PR?

**Comment by ankit-varma10** [comment]
@GabrielBianconi Yes, I'm done with the development, but I need to investigate some of the tooling options. I should be done with this by Wed (01/29).

Does this sound good?

**Comment by GabrielBianconi** [comment]
Sounds good, thanks

**Comment by ankit-varma10** [comment]
Created PR [#880](https://github.com/tensorzero/tensorzero/pull/880)
